### ISSUE TITLE
Worked exited (code: 15) error on Linux host

### ISSUE THREAD

**Issue by ddelnano** [rezzza/docker-flow#12]
I have been using this docker image since the 0.26.0 tag.  When you released the 0.29.0 and 0.30.0 tags I wanted to upgrade to use newer flow features.  For our projects at work most devs use Macs and the 0.29.0 and 0.30.0 run fine, however, on our CI server we get the following output.

<img width="1404" alt="screen shot 2016-08-01 at 9 42 51 am" src="https://cloud.githubusercontent.com/assets/5855593/17295943/c1408b3e-57cc-11e6-9a42-8b279b3652da.png">

Our CI server is running docker version 1.11.2 and I have successfully run those tags on my Mac with docker versions 1.11.1 and 1.11.2.

I am wondering if you use these images on a Mac or have run into issues with this error before?

**Comment by shouze** [comment]
Hi @ddelnano I wonder if it could be a docker issue more than a flow issue.

Can you please try this one and check the return code again?
```shell
docker run --rm -it --entrypoint 'ash -c "flow check"' -v YOUR_LOCAL_PATH:YOUR_CONTAINER_PATH -w YOUR_CONTAINER_PATH rezzza/docker-flow:0.30.0
```

We're running the command as you do on travis CI therefore and it works for any release of the containerized flow but I've already seen issues about docker run/exec and return codes so I wait for your feedback on the command I've suggested before.

**Comment by ddelnano** [comment]
@shouze I get the following.  What is `ash`?  Is that a different shell?

```
docker: Error response from daemon: Container command 'ash -c "flow check"' not found or does not exist..
```

**Comment by ddelnano** [comment]
Also I was thinking this could be a docker issue as well.

**Comment by shouze** [comment]
@ddelnano yes `ash` is a different shell as the container we build is based on Alpine Linux (lightweight Linux distribution).

**Comment by ddelnano** [comment]
Ok so that entrypoint command must be a little weird than because I am able to get an `ash` shell prompt if I just change the entrypoint to `ash`.

I receive the same output if I change the entrypoint to `ash` and run flow check.  Only difference is this command never exits so I have to do <C-c> to exit it.

<img width="468" alt="screen shot 2016-08-01 at 10 55 14 am" src="https://cloud.githubusercontent.com/assets/5855593/17298104/9bbbc324-57d6-11e6-9fa1-61b97a869226.png">

**Comment by ddelnano** [comment]
Do you know what docker version you are using on your Travis builds?

**Comment by shouze** [comment]
@ddelnano ok so looking at https://github.com/rezzza/docker-flow/issues/12#issuecomment-236605365 it's probably a flow issue. I don't know exactly what kind if issue at the moment.

We're using docker `1.12.0~rc4-0~trusty` on ubuntu trusty ATM on travis CI (we have to do the switch to the stable 1.12 as it's been released few days ago).

**Comment by ddelnano** [comment]
Ok I am going to try and update our server to the 1.12 release and see if that has any effect.  Thank you for your help so far.

**Comment by shouze** [comment]
@ddelnano you're welcome, keep me up to date.

**Comment by MikaFima** [comment]
I'm currently experiencing the same experience on Ubuntu trusty.

Here is my docker's check-config.sh results: https://gist.github.com/MikaFima/167c987fe01119693cee575f4b554a77

I should be glad that we find a solution for updating from 0.26.

**Comment by shouze** [comment]
Thx @MikaFima! Ping @ddelnano could you run `sudo bash -c /usr/share/docker-engine/contrib/check-config.sh` on your jenkins machine too? Thx.
I will compare with travis-ci one.

We've tested with @MikaFima if it could come from using shared volumes or something like that by creating a derived container with the project code & modules copied into the new container and we stuck with this error code 15 so it's probably some docker config and/or Linux running system difference.

**Comment by ddelnano** [comment]
Here is a gist of my `sudo bash -c /usr/share/docker-engine/contrib/check-config.sh` on my jenkins server. https://gist.github.com/ddelnano/9a06f2e0a2de691d9c3481a6354ff33f

I still have not been able to try the upgrade and probably won't for some time.

**Comment by shouze** [comment]
ok thx @ddelnano it could be related to apparmor and/or seccomp (enabled for both you & @MikaFima). We will test tomorrow on @MikaFima by disabling apparmor & seccomp as explained in https://github.com/docker/docker/issues/7276#issuecomment-194149208. I will check also on travis, I keep you updated ;)

**Comment by shouze** [comment]
Ok @MikaFima @ddelnano  I confirm that's probably caused by apparmor as this is the only diff between all `check-config.sh` comparisons: apparmor is not running on travis ci ubuntu trusty image.

**Comment by ddelnano** [comment]
@shouze from the link you posted above I tried the following still with no luck.
```bash
docker run --rm -it -v $(pwd):/app --security-opt=apparmor:unconfined --security-opt=seccomp:unconfined rezzza/docker-flow:0.30.0
```

I tried disabling / removing apparmour from my server (so I think) and the check-config.sh still shows it enabled.

**Comment by ddelnano** [comment]
@shouze did you get a chance to try it with apparmor & seccomp disabled?

**Comment by shouze** [comment]
@ddelnano no, still investigating

**Comment by briancavalier** [comment]
Hey everyone, thanks for all the work investigating this so far.  I work with @ddelnano, and am just wondering if anyone's uncovered more info.

**Comment by shouze** [comment]
Hi, sorry we still have the issue on Linux Ubuntu machines... I will try to investigate more by the end of the week. I'm running myself MacOS so I don't experiment the issue on a daily basis sorry, will see with @MikaFima soon and keep you updated.

**Comment by briancavalier** [comment]
Thanks @shouze.  I haven't kept up with everything that @ddelnano has tried so far in helping to debug this, but is there anything else we can do to help?

**Comment by shouze** [comment]
@briancavalier @ddelnano ok guys we have the fix, you have to mount a tmpfs directory in fact, plz try that kind of command to confirm:
```
docker run --rm -it -v $(pwd):/app --tmpfs /tmp rezzza/docker-flow:0.30.0
```

**Comment by ddelnano** [comment]
hm now i am seeing a different error.

```
[2017-02-10 23:43:20] We've run out of filesystems to use for shared memory
```

Googling around on that showed that it is related to error code 15 which is the issue I had originally.  I tried 0.30.0 and 0.32.0, same result.

How did you come to the conclusion that `--tmpfs` will help?

**Comment by shouze** [comment]
ok bad news... flow 0.32 is kind of outdated now. I admit we've tested that with another docker image, another image of our own: `rezzza/docker-node:karibbu-3.4.0`.

Do you think you can try with this one? You should have to run something like
```
docker run --rm -it -v $(pwd):/app --tmpfs /tmp rezzza/docker-node:karibbu-3.4.0 flow check
```

It's working for us on @MikaFima machine with docker >= 1.13.0 on a pretty old ubuntu (I guess it's trusty, he will be able to confirm that to you soon).

**Comment by ddelnano** [comment]
I tried that and got the exact same error message :(

**Comment by ddelnano** [comment]
@briancavalier @shouze I found the solution!

I was working on running Selenium in a docker container and saw the recommendation to bind mount the host's `/dev/shm` into the container.

So the following works.

```
docker run --rm -it -v $(pwd):/app -v /dev/shm:/dev/shm rezzza/docker-node:karibbu-3.4.0 flow check
```

**Comment by ddelnano** [comment]
Also @shouze wanted to thank you for all the debugging you helped with.

**Comment by briancavalier** [comment]
🎉 Awesome.  Thanks for all the work here @ddelnano and @shouze.

**Comment by shouze** [comment]
@ddelnano so nice! Thank you for your patience, so yes I've added a troubleshooting note into the README and... as a bonus I've published a new tagger container for flow 0.39.0 😉  see #26

**Comment by ddelnano** [comment]
Sweet can't wait to start updating our stuff 🤙🏻
